pkg/cli/admin/upgrade/upgrade.go

```diff
@@ -221,23 +221,15 @@ func (o *Options) Run() error {
 				fmt.Fprintf(o.Out, "info: Cluster is already at version %s\n", o.To)
 				return nil
 			}
-			for _, available := range cv.Status.AvailableUpdates {
-				if available.Version == o.To {
-					update = &configv1.Update{
-						Version: available.Version,
-						Image:   available.Image,
-					}
-					break
-				}
-			}
+			update = findUpdateFromConfigVersion(cv, o.To)
 			if update == nil {
 				if len(cv.Status.AvailableUpdates) == 0 {
 					if c := findCondition(cv.Status.Conditions, configv1.RetrievedUpdates); c != nil && c.Status == configv1.ConditionFalse {
 						return fmt.Errorf("Can't look up image for version %s. %v", o.To, c.Message)
 					}
 					return fmt.Errorf("No available updates, specify --to-image or wait for new updates to be available")
 				}
-				return fmt.Errorf("The update %s is not one of the available updates: %s", o.To, strings.Join(versionStrings(cv.Status.AvailableUpdates), ", "))
+				return fmt.Errorf("The update %s is neither one of the available updates (%s) nor a history entry", o.To, strings.Join(versionStrings(cv.Status.AvailableUpdates), ", "))
```

Comment:
Why can not we just say `The %s is not available to update`? Do we really need to print the available update versions?

Reply:
I not comfortable in printing about `nor a history entry` when we do not encourage customers to downgrade.

Reply:
Should we put this behind `--allow-versions-from-history` or some such with a warning when it's needed, like our other `--allow-...` handling?

Reply:
But is a version from history still necessarily available in channel for an update?

Reply:
Good point from both of you about having a guard around downgrades.  I've pushed 2b9210665 -> adb05a76b to wire this up to the existing `--allow-explicit-upgrade`.  If `--to ...` picks a value from `availableUpdates`, it doesn't matter whether you've set `--allow-explicit-upgrade` or not.  But with adb05a76b, when `--to ...` picks a value from the history, you will either fail to update (when `--allow-explicit-upgrade` was not set) or successfully trigger the update but with a stderr complaint (when `--allow-explicit-upgrade` was set).  I think that addresses the safety concerns.

> Do we really need to print the available update versions?

No, but mentioning them means folks can skip a separate `oc adm upgrade` step that may return a different list of available updates than the set we just used to reject their choice.

> I not comfortable in printing about `nor a history entry`...

Does the new `--allow-explicit-upgrade` integration address your concerns there?

> But is a version from history still necessarily available in channel for an update?

Nope.  And the point of this PR is allowing these updates to previous versions without requiring either `oc patch ...` or folks looking up the by-digest pullspec themselves.  But yes, there is a safety issue with rolling the cluster back.  Does the new `--allow-explicit-upgrade` integration address your concerns there?

Reply:
@wking `nor a history entry` , will the customer understand what this means? If yes then I am ok with this one.

Reply:
> @wking nor a history entry , will the customer understand what this means?

Are you asking for a rephrase that says "you can draw from either set" without using "neither ... nor ..."?  Or something that mentions the fact this history entries may not be recommended updates?  Or...?

Reply:
Here is my suggestion `The update %s is neither one of the available updates (%s) nor present in the update history`

Reply:
> Here is my suggestion...

Rebased onto master and updated to use your text with adb05a76b -> 754657b01.